python/paddle/fluid/parallel_executor.py

```diff
@@ -50,10 +52,18 @@ def __init__(self,
             else:
                 min(len(self._places) * 2, multiprocessing.cpu_count())
 
-        startup = framework.default_startup_program()
-        main = framework.default_main_program()
+        main = main_program
+        main = main if main else framework.default_main_program()
         scope = executor.global_scope()
 
+        local_scopes = share_vars_from.executor.local_scopes(
+        ) if share_vars_from else []
+
+        vars = [
```

Comment:
call it persistable_vars? Does this include all vars needed? Or maybe it includes more vars than necessary? We can leave a TODO here so that we can have the right var collection here in the future?

Reply:
>  Or maybe it includes more vars than necessary

Now, this case is existing. The C++ code filters the non LoDTensor.